pkg/controller/framework/controller.go

```diff
@@ -106,6 +106,16 @@ func (c *Controller) HasSynced() bool {
 	return c.reflector.LastSyncResourceVersion() != ""
 }
 
+// Requeue adds the provided object back into the queue if it does not already exist.
+func (c *Controller) Requeue(obj interface{}) error {
```

Comment:
minor nit: double check that obj is not already a cache.Deltas (or Delta) object before stuffing it into one.